echopype/echodata/combine.py

```diff
@@ -46,12 +46,12 @@ def check_zarr_path(
         If ``zarr_path`` already exists and ``overwrite=False``
     """
 
-    # check that zarr_path is a string
-    if not isinstance(zarr_path, str):
+    # check that zarr_path is a string, if it is not None
+    if (zarr_path is not None) and (not isinstance(zarr_path, str)):
         raise TypeError(f"zarr_path must be of type {str}")
 
-    # check that the appropriate suffix was provided
-    if not zarr_path.strip("/").endswith(".zarr"):
+    # check that the appropriate suffix was provided, if zarr_path is not None
+    if (zarr_path is not None) and (not zarr_path.strip("/").endswith(".zarr")):
         raise ValueError("The provided zarr_path input must have '.zarr' suffix!")
```

Comment:
Seems like this should first include a check of whether `zarr_path` is None?
```python
if zarr_path is not None:
    # check that zarr_path is a string
    if not isinstance(zarr_path, str):
        raise TypeError(f"zarr_path must be of type {str}")

    # check that the appropriate suffix was provided
    if not zarr_path.strip("/").endswith(".zarr"):
        raise ValueError("The provided zarr_path input must have '.zarr' suffix!")
else:
   raise ValueError("A zarr_path must be provided as input!")
```

Reply:
I am pretty sure the logic is the same, unless I am missing something? In the code you provide, we do not want to error out if `zarr_path is None`. If it is `None`, we simply create the path for them in the `validate_output_path` function.

Reply:
> If it is None, we simply create the path for them in the validate_output_path function.

Oh right somehow I forgot about that -- then we can remove the `else:` part.

Yes the logic is the same, but I thought it is cleaner to keep the same `if zarr_path is not None` component at a higher level, so the check is done once and for readability.